gcloud.go

```diff
@@ -106,60 +117,105 @@ func createVMs(names []string, opts VMOpts) error {
 	}
 	defer os.Remove(filename)
 
-	// Fixed args.
-	args := []string{
-		"compute", "instances", "create",
-		"--subnet", "default",
-		"--maintenance-policy", "MIGRATE",
-		"--service-account", "21965078311-compute@developer.gserviceaccount.com",
-		"--scopes", "default,storage-rw",
-		"--image", "ubuntu-1604-xenial-v20171002",
-		"--image-project", "ubuntu-os-cloud",
-		"--boot-disk-size", "10",
-		"--boot-disk-type", "pd-ssd",
+	if !opts.GeoDistributed {
+		zones = []string{zones[0]}
 	}
 
-	// Dynamic args.
-	if opts.UseLocalSSD {
-		args = append(args, "--local-ssd", "interface=SCSI")
-	}
-	args = append(args, "--machine-type", opts.MachineType)
-	args = append(args, "--labels", fmt.Sprintf("lifetime=%s", opts.Lifetime))
+	totalNodes := float64(len(names))
+	totalZones := float64(len(zones))
+	nodesPerRegion := int(math.Ceil(totalNodes / totalZones))
 
-	args = append(args, "--metadata-from-file", fmt.Sprintf("startup-script=%s", filename))
-	args = append(args, "--project", project)
-	args = append(args, "--zone", zone)
-	args = append(args, names...)
+	ct := int(0)
+	i := 0
 
-	cmd := exec.Command("gcloud", args...)
+	var wg sync.WaitGroup
+	wg.Add(len(zones))
+
+	for i < len(names) {
+		// Fixed args.
+		args := []string{
```

Comment:
you can keep the common args outside the loop as long as you use a new per-iteration arg. eg:
```
args := []string{ ....}
for .... {
  zoneArgs = append(args, newArgs...)
  run(gcloud, zoneArgs)
}
```